messages/pl/yii-validator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+return [
+    'The data must have at least "{min}" filled attributes.' => 'Dane muszą mieć co najmniej "{min}" wypełnionych atrybutów.',
+    'Value must be an array or an object.' => 'Wartość musi być tablicą lub obiektem.',
+    'The allowed types are integer, float, string, boolean. {type} given.' => 'Dozwolone typy to integer, float, string, boolean. Podano {type}.',
+    'Value must be either "{true}" or "{false}".' => 'Wartość musi być albo "{true}" albo "{false}".',
+    'The allowed types are integer, float, string, boolean and null.' => 'Dozwolone typy to integer, float, string, boolean i null.',
+    'The attribute value returned from a custom data set must have a scalar type.' => 'Wartość atrybutu zwrócona z niestandardowego zestawu danych musi być typu skalarnego.',
+    'Value must be equal to "{targetValueOrAttribute}".' => 'Wartość musi być równa "{targetValueOrAttribute}".',
+    'Value must not be equal to "{targetValueOrAttribute}".' => 'Wartość nie może być równa "{targetValueOrAttribute}".',
+    'Value must be greater than "{targetValueOrAttribute}".' => 'Wartość musi być większa niż "{targetValueOrAttribute}".',
+    'Value must be greater than or equal to "{targetValueOrAttribute}".' => 'Wartość musi być większa lub równa od "{targetValueOrAttribute}".',
+    'Value must be less than "{targetValueOrAttribute}".' => 'Wartość musi być mniejsza niż "{targetValueOrAttribute}".',
+    'Value must be less than or equal to "{targetValueOrAttribute}".' => 'Wartość musi być mniejsza lub równa od "{targetValueOrAttribute}".',
+    'This value must be an array or implement \Countable interface.' => 'Ta wartość musi być tablicą lub implementacją interfejsu \Countable.',
+    'This value must contain at least {min, number} {min, plural, one{item} other{items}}.' => 'Ta wartość musi zawierać co najmniej {min, number} {min, number} {min, plural, one{element} few{elementy} many{elementów} other{elementu}}.',
+    'This value must contain at most {max, number} {max, plural, one{item} other{items}}.' => 'Ta wartość musi zawierać co najwyżej {max, number} {max, plural, one{element} few{elementy} many{elementów} other{elementu}}.',
+    'This value must contain exactly {exactly, number} {exactly, plural, one{item} other{items}}.' => 'Ta wartość musi zawierać dokładnie {exactly, number} {exactly, plural, one{element} few{elementy} many{elementów} other{elementu}}.',
+    'Value must be array or iterable.' => 'Wartość musi być tablicą lub iteracją.',
+    'Every iterable key must have an integer or a string type.' => 'Każdy klucz iterowalny musi być liczbą lub tekstem.',
+    'The value must have a string type.' => 'Wartość musi być typu tekstowego.',
```

Comment:
Hm, this is used in context of validators: Json, Url, Ip and Email. Word "string" here is correct and understable for polish programmers, but this validation error message will be show also to simple user, and he probably ask himself a question "Co to jest string?"
Or not?

Reply:
I was wondering as well why are there so many similar messages like this one and "Value should be string" for example. This looks like a design issue. Here I've seen different construct with "have" so decided it's not about being something ("być") but rather having something ("mieć") but I'm not sure. @arogachev maybe you can shed a light on this? And as for the end-user not understanding it, what about other messages where we are throwing floats, nulls, and so on?

Reply:
@tomaszkane Thanks for adding the translations and providing feedback.

@bizley Thanks for helping with review and providing feedback.

To answer your questions:

1) Grouping of similar messages indeed needs to be done. Created the issue for it - #525.

2) As for error messages with technical details - it's expected due to intent for them to be more specific. They are OK for APIs, and with forms should never be shown with correct configuration on the client side. In the latter case, even when frontend configuration is incorrect, they can be useful for developers during development / testing. 

And there is an option to override them to something more broad. The problem with that approach and current translations' mapping is that it can affect multiple rules though. We might improve this with using keys instead, something like "json.incorrectInputMessage', but decided to wait for feedback first. Created the issue for it too - #526.

These problems can be further discussed in these issues' comments. But for now I suggest to choose the translations that would be more correct in your opinion (can be fine tuned later) based on available source messages. Similar translations can be removed later by us without involving the translators.

Reply:
Ok, let's leave it as it is now then. Thanks.